src/AutoBuild.jl

```
@@ -856,47 +936,70 @@ end
     end
 
     if !isdir(code_dir)
-        # If it does exist, clone it down:
-        @info("Cloning wrapper code repo from https://github.com/$(deploy_repo) into $(code_dir)")
-        Wizard.with_gitcreds(gh_username, gh_auth.token) do creds
-            LibGit2.clone("https://github.com/$(deploy_repo)", code_dir; credentials=creds)
+        if gh_repo === nothing
+            # The gh repo didn't exist and we didn't want to create it. Initialize an empty repository
+            mkpath(code_dir)
+            repo = LibGit2.init(code_dir)
+            LibGit2.commit(repo, "Initial empty commit")
+        else
+            # If it does exist, clone it down:
+            @info("Cloning wrapper code repo from https://github.com/$(deploy_repo) into $(code_dir)")
+            with_gh_auth() do creds
+                LibGit2.clone("https://github.com/$(deploy_repo)", code_dir; credentials=creds)
+            end
         end
     else
         # Otherwise, hard-reset to latest master:
         repo = LibGit2.GitRepo(code_dir)
-        Wizard.with_gitcreds(gh_username, gh_auth.token) do creds
-            LibGit2.fetch(repo; credentials=creds)
-        end
-        origin_master_oid = LibGit2.GitHash(LibGit2.lookup_branch(repo, "origin/master", true))
-        LibGit2.reset!(repo, origin_master_oid, LibGit2.Consts.RESET_HARD)
-        if string(LibGit2.head_oid(repo)) != string(origin_master_oid)
-            LibGit2.branch!(repo, "master", string(origin_master_oid); force=true)
+        try
+            with_gh_auth() do creds
+                LibGit2.fetch(repo; credentials=creds)
+            end
+            origin_master_oid = LibGit2.GitHash(LibGit2.lookup_branch(repo, "origin/master", true))
+            LibGit2.reset!(repo, origin_master_oid, LibGit2.Consts.RESET_HARD)
+            if string(LibGit2.head_oid(repo)) != string(origin_master_oid)
+                LibGit2.branch!(repo, "master", string(origin_master_oid); force=true)
+            end
+        catch
```

I don't really like big `try/catch` chunks like this with no error handling; what is the error you're trying to ignore?

---

The repo may not have an upstream.

---

You mean in the event that it was previously created during a build with `--deploy=universe`?  Can we test for the existence of the `origin/master` branch and wrap this in an `if` then?  I don't want to swallow unrelated errors.

---

sure